### PR TITLE
Add  functions freeze and unfreeze

### PR DESCRIPTION
**Pull request by MichaelKim20**
I added a function to the class wallet to make freezing and unfreezing easier.

### CODE REVIEW

File: src/modules/utils/UTXOManager.ts [hunk 1]
```diff
@@ -276,6 +276,28 @@ export class UTXOProvider {
...
+        let amount = JSBI.BigInt("400000000000");
+        while (true) {
+            let additional = await this.client.getWalletUTXOs(
+                this.address,
+                amount,
+                BalanceType.frozen,
+                result.length === 0 ? undefined : result[result.length - 1].utxo
+            );
+
+            if (additional.length == 0) break;
```
Comment (TrustHenry) [inline src/modules/utils/UTXOManager.ts:295]:
```suggestion
            if (additional.length === 0) break;
```

**Comment by TrustHenry** [comment]
Conflict occurred due to previous PR merge.

**Comment by MichaelKim20** [comment]
@TrustHenry I rebased it

File: src/modules/wallet/Wallet.ts [hunk 2]
```diff
@@ -507,4 +533,94 @@ export class Wallet {
...
+        let frozenUtxos: UnspentTxOutput[];
+        try {
+            frozenUtxos = await this.frozenUtxoProvider.getUTXO(amount);
+        } catch (e) {
+            return { code: WalletResultCode.FailedRequestUTXO, message: e.message };
+        }
+
+        return {
+            code: WalletResultCode.Success,
+            message: "Success.",
```
Comment (TrustHenry) [inline src/modules/wallet/Wallet.ts:622]: How do I distinguish it from the error message?

**Review by TrustHenry — APPROVED** [review]
It's good that the error code is well defined.

File: src/modules/wallet/Wallet.ts [hunk 3]
```diff
@@ -226,10 +236,16 @@ export class Wallet {
...
      * The payload is the data to be stored in the block. A separate cost is incurred.
+     * @param output_type The type of transaction output (0: OutputType.Payment, 1: OutputType.Freeze)
      * @param receiver  The array of recipient information
      * @param payload   The data to be stored, not used if not entered.
+     * @private
      */
-    public async transfer(receiver: IWalletReceiver[], payload?: Buffer): Promise<IWalletResult> {
+    private async _transfer(
+        output_type: OutputType,
+        receiver: IWalletReceiver[],
```
Comment (TrustHenry) [inline src/modules/wallet/Wallet.ts:246]: Please check the empty array value.

Comment (MichaelKim20) [reply src/modules/wallet/Wallet.ts:246]: I think we should add another PR.

**Pull request merged**